### PR TITLE
Fix app/bot names

### PR DESCRIPTION
**Pull request by plasne**
THIS PR ADDRESSES THESE ISSUES:

- If you have a 1:1 chat with a bot and it doesn't reply, the thread will show with your name, not the bot name. This is because we are only getting the bot info from the lastMessagePreview.
- From Graph API/notify the displayName is the name of the application for the bot, not the name of the App (as it is in Teams).

To address these issues, we need to call this API for each chat thread:
[List apps in chat - Microsoft Graph v1.0 | Microsoft Learn](https://learn.microsoft.com/en-us/graph/api/chat-list-installedapps?view=graph-rest-1.0&tabs=http#request-1)

We need to expand for bot info. If one or more bots exist, we should use those for the name. In addition, we should cache the bot ID and name so that when we see application displayNames we replace them with the Teams App name.

A FEW NOTES ABOUT THE IMPLEMENTATION:

- This loads the app data on-demand when it is required.
- The `useRef` implementation ensures only 1 load can happen.
- Additional permissions are required for this to work.

### CODE REVIEW

**Comment by MattBillFred** [comment]
I see a rare condition where the bot address is showing instead of the bot name.

To duplicate
1. Clear out the lastRead cache.
2. Refresh.
3. Wait for the bot's address to be replaced with the bot's name.
4. Click on the bot, the name will be replaced with the address.

After clearing cache and refreshing:
chrome_sefNNBBDAX

After clicking on bot name:
chrome_omzzOmqvGb

File: packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx [hunk 1]
```diff
@@ -161,41 +194,98 @@ export const ChatListItem = ({ chat, myId, isSelected, isRead }: IMgtChatListIte
     return <></>;
   }
 
+  const getMemberName = (member: ConversationMember): string => {
+    return member?.displayName || (member as AadUserConversationMember)?.email || member?.id || 'Uknown';
```
Comment (MattBillFred) [inline packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx]:
```suggestion
    return member?.displayName || (member as AadUserConversationMember)?.email || member?.id || 'Unknown';
```

**Comment by plasne** [comment]
> I see a rare condition where the bot address is showing instead of the bot name.
> 
> To duplicate
> 
> 1. Clear out the lastRead cache.
> 2. Refresh.
> 3. Wait for the bot's address to be replaced with the bot's name.
> 4. Click on the bot, the name will be replaced with the address.
> 
> After clearing cache and refreshing: chrome_sefNNBBDAX
> 
> After clicking on bot name: chrome_omzzOmqvGb

Awesome find. I am able to reproduce. I have no idea what could even cause that, but I will continue looking.

**Comment by plasne** [comment]
I found the issue and fixed it, though it did also spark discussion about another change we should do.

**Review by andrewDoing — APPROVED** [review]

**Review by MattBillFred — APPROVED** [review]

**Pull request merged**